Source/inv.cpp

```diff
@@ -432,65 +440,77 @@ void CheckInvPaste(int pnum, Point cursorPosition)
 			}
 		};
 		inv_body_loc slot = iLocToInvLoc(il);
-		NetSendCmdChItem(false, slot);
-		if (player.InvBody[slot].isEmpty())
-			player.InvBody[slot] = player.HoldItem;
-		else
-			cn = SwapItem(player.InvBody[slot], player.HoldItem);
+		Item previouslyEquippedItem = player.InvBody[slot];
+		ChangeEquipment(player, slot, player.HoldItem);
+		if (!previouslyEquippedItem.isEmpty()) {
+			player.HoldItem = previouslyEquippedItem;
+			cn = previouslyEquippedItem._iCurs + CURSOR_FIRSTITEM;
+		}
 		break;
 	}
 	case ILOC_ONEHAND:
 		if (r <= SLOTXY_HAND_LEFT_LAST) {
 			if (player.InvBody[INVLOC_HAND_LEFT].isEmpty()) {
 				if ((player.InvBody[INVLOC_HAND_RIGHT].isEmpty() || player.InvBody[INVLOC_HAND_RIGHT]._iClass != player.HoldItem._iClass)
 				    || (player._pClass == HeroClass::Bard && player.InvBody[INVLOC_HAND_RIGHT]._iClass == ICLASS_WEAPON && player.HoldItem._iClass == ICLASS_WEAPON)) {
-					NetSendCmdChItem(false, INVLOC_HAND_LEFT);
-					player.InvBody[INVLOC_HAND_LEFT] = player.HoldItem;
+					ChangeEquipment(player, INVLOC_HAND_LEFT, player.HoldItem);
 				} else {
-					NetSendCmdChItem(false, INVLOC_HAND_RIGHT);
-					cn = SwapItem(player.InvBody[INVLOC_HAND_RIGHT], player.HoldItem);
+					Item previouslyEquippedItem = player.InvBody[INVLOC_HAND_RIGHT];
+					ChangeEquipment(player, INVLOC_HAND_RIGHT, player.HoldItem);
+					player.HoldItem = previouslyEquippedItem;
+					cn = previouslyEquippedItem._iCurs + CURSOR_FIRSTITEM;

```

Comment:
these 4 lines repeat a lot, maybe worth making some function for that?

Reply:
I do think that was the idea behind `SwapItem()`, but I found the pattern significantly more confusing because of the return value. Rather than introducing a function for these four lines, I'd rather see a cleanup of `CheckInvPaste()` that deduplicates all the nested conditionals in the `ILOC_ONEHAND` case, which should clear up basically all of this duplication as well. I had to resist the urge to do that cleanup in this PR, as I figured it was more of a 1.5.0 thing.

Perhaps I'll submit another PR based on top of this one so you can see what I'm talking about.

Reply:
I had the same though but also came to the same conclusion

Reply:
FYI, #4246